sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NameOfShouldBeUsed.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public void Method_01(int arg1, int arg2)
+        {
+            if (arg1 < 0)
+                throw new Exception("arg1"); // Noncompliant {{Replace the string 'arg1' with 'nameof(arg1)'.}}
```

Comment:
I am -1 on handling this case. I would handle only exceptions inheriting from `ArgumentException` because we have no idea on the other cases whether it is expected or not.

Reply:
I would also check only the known ArgumentExceptions, because custom descendants could have different ctor signatures...